File: app/models/icd_code.rb
require 'abstract/categorised_code'
require 'catalog'

class IcdCode < CategorisedCode

  def name
    return 'IcdCode'
  end

end

# Adds an extension method to code catalog
# annotated with _code pragma
# to indicate that chop code exists
class Catalog
  pragmatize!

  _code
  def icd_code_in(aContext)
    IcdCode
  end
end

File: app/assets/javascripts/models/codes/IcdCode.js
define([
    'models/abstract/CategorisedDataset'
], function(
    CategorisedDataset
) {

    function IcdCode() {
        var _this = new CategorisedDataset();


        /**
         * @returns {string}
         */
        _this.name = function() {
            return 'Icd Code Dataset';
        };

        return _this;
    }

    IcdCode.ID = 'icd';
    return IcdCode;
});